### DISCUSSION TITLE
How to rate limit with Coolify?

### DISCUSSION THREAD

**Discussion by AlejandroAkbal** [coollabsio/coolify#1817] [Q&A]
I'm pretty new to both Coolify and Traefik, is there any way I can rate limit HTTP requests to some projects?

**Comment by mrtncode** [comment]
+1
Same problem here. Any solution?

**Comment by djsisson** [comment]
add the following middleware, either in a dynamic file, and reference it, or directly via labels:

https://doc.traefik.io/traefik/middlewares/http/ratelimit/

**Reply by mrtncode** [reply]
> add the following middleware, either in a dynamic file, and reference it, or directly via labels:

@djsisson Thank you very much for your answer! What exactly do you mean by dynamic file? How do you do that in coolify? Or how do you add labels? is that possible directly in coolify or do I have to use my project in a docker compose file?

**Reply by djsisson** [reply]
the link above gives example of using dynamic(yaml) file or labels, if you use file, then you only need to add one label to your app/compose

```coolify.traefik.middlewares=test-ratelimit@file```
then add the dynamic file (see proxy page in coolify)
```yaml
http:
  middlewares:
    test-ratelimit:
      rateLimit:
        average: 100
        burst: 200
```
The ```test-ratelimit``` needs to match in both the label and the name of your middleware (it can be anything as long as it doesn't conflict with any other middleware name)

if you don't want to use a dynamic file 

then you would just add the labels either to your compose or in the labels section if this is an app

```
  - "traefik.http.middlewares.test-ratelimit.ratelimit.average=100"
  - "traefik.http.middlewares.test-ratelimit.ratelimit.burst=200"
```

ps using a dynamic file makes it way easier to test, since you can live change the file values and get instant changes in your app, for label changing it requires a redeploy

**Reply by mrtncode** [reply]
Many thanks! I got it now :)
